python/GafferUI/ColorChooser.py

```diff
@@ -611,6 +614,31 @@ def setErrored( self, errored ) :
 	def getErrored( self ) :
 		return any( w.getErrored() for w in self.__numericWidgets.values() )
 
+	def setVisibleComponents( self, components ) :
+
+		self.__setVisibleComponentsInternal( components )
+
+	def getVisibleComponents( self ) :
+
+		return "".join( [ k for k, v in self.__sliders.items() if v.getVisible() ] )
+
+	def setStaticComponent( self, component ) :
+
+		self.__setStaticComponentInternal( component )
+
+	def getStaticComponent( self ) :
+
+		c, staticComponent = self.__colorField.getColor()
+		return staticComponent
+
+	def setColorFieldVisible( self, visible ) :
+
+		self.__setColorFieldVisibleInternal( visible )
+
+	def getColorFieldVisible( self ) :
+
+		return self.__colorField.getVisible()
```

Comment:
I did wonder if we could have just one property here instead of two, and use `None` to mean invisible. But maybe that doesn't work if we want the menu to have separate options for visibility and component...